### DISCUSSION TITLE
How to get subgen to pick the updated .env variables without restarting the docker container?

### DISCUSSION THREAD

**Discussion by kuddukan1** [McCloudS/subgen#92] [Q&A]
Is it possible to get subgen to use the updated .env variables without restarting the docker container?  Say I enter a new transcribe_folders path and then click the "Save as subgen.env and reload" button.  Subgen doesn't appear to scan the new folders until I stop and start the docker container.

**Comment by McCloudS** [comment]
I just tried adding that as a feature.  Give it a couple minutes and download the new one (or set UPDATE=True).  It should run
        transcribe_existing(transcribe_folders) when the .env variables are saved/reloaded.

**Comment by kuddukan1** [comment]
subgen  | Environment variables have been loaded from subgen.env
subgen  | Downloading subgen.py from GitHub branch main...
subgen  | File downloaded successfully to subgen.py
subgen  | Launching subgen.py
subgen  | Traceback (most recent call last):
subgen  |   File "/subgen/subgen.py", line 116, in <module>
subgen  |     update_env_variables()
subgen  |   File "/subgen/subgen.py", line 114, in update_env_variables
subgen  |     transcribe_existing(transcribe_folders)
subgen  | NameError: name 'transcribe_existing' is not defined
subgen  | Traceback (most recent call last):
subgen  |   File "/subgen/launcher.py", line 168, in <module>
subgen  |     main()
subgen  |   File "/subgen/launcher.py", line 163, in main
subgen  |     subprocess.run([f'{python_cmd}', '-u', 'subgen.py'], check=True)
subgen  |   File "/usr/lib/python3.10/subprocess.py", line 526, in run
subgen  |     raise CalledProcessError(retcode, process.args,
subgen  | subprocess.CalledProcessError: Command '['python3', '-u', 'subgen.py']' returned non-zero exit status 1.
subgen exited with code 1

**Comment by kuddukan1** [comment]
I got the result above when I stopped the container, pulled, and started.  I stopped the container, did a docker system prune, and now the container is up and running again.  I also frequently seem to have to docker system prune to get it going again after I have changed the ENV parameters.

**Comment by McCloudS** [comment]
I think I goofed a line, made a change, let me know if it works.

**Reply by kuddukan1** [reply]
when clicking "reload" button do you expect subgen to stop what's it doing and restart with the new parameters?  mine appears to be continuing the old jobs

**Reply by kuddukan1** [reply]
maybe my problem is that I have ENV variables set in the docker compose file, and also using the localhost:9000 to set them.

**Reply by McCloudS** [reply]
Try setting your transcribe folders in your compose to blank (or just deleting it) and see if that works.  They’ll compete with each other in odd ways.

**Reply by McCloudS** [reply]
It will finish the queue that it has added to that point before starting your new set of folders.  You can also just use the pipe separator as outlined in the readme to add multiple folder paths in one go.

**Comment by McCloudS** [comment]
Reverted to remove this until I have more time to test it.  It had unintended impacts on variables.
